src/components/user/repo/user.repo.js

```diff
@@ -0,0 +1,32 @@
+const { userModel: User } = require('../../../services/user/model');
+
+const getUser = async (id, app) => {
+	return app.service('users').get(id);
+};
+
+const replaceUserWithTombstone = async (id, replaceData = {}, app) => {
+	const user = await getUser(id, app);
```

Comment:
app.service('users').get(id) return a lot of stuff that is not releated, it resolve roles and permissions, birthday and so on and add this as keys. Some of this stuff as model after hook, some as hook. It is to much to save it in trashbin.

Otherwise it is called on this position and you return the Model result. But is this the result of the old data or the new and you do not use lean(), also you return the mongoose schema. That should not pass out of the repo layer.

Reply:
You also pass out the error,is it what we want for this db action?

Reply:
changed to usersModel